src/main/scala/com/gu/invoicing/nextinvoicedate/Impl.scala

```diff
@@ -0,0 +1,62 @@
+package com.gu.invoicing.nextinvoicedate
+
+import java.time.LocalDate
+import com.gu.invoicing.common.ZuoraAuth.{accessToken, zuoraApiHost}
+import com.gu.invoicing.nextinvoicedate.Model._
+import scalaj.http.Http
+import scala.util.chaining._
+import pprint._
+
+object Impl {
+  def getAccountId(name: String): String = {
+    Http(s"$zuoraApiHost/v1/subscriptions/$name")
+      .header("Authorization", s"Bearer $accessToken")
+      .asString
+      .body
+      .pipe(read[Subscription](_))
+      .accountId
+  }
+  def getBillingPreview(accountId: String): List[InvoiceItem] = {
+    Http(s"$zuoraApiHost/v1/operations/billing-preview")
+      .header("Authorization", s"Bearer $accessToken")
+      .header("Content-Type", "application/json")
+      .postData(
+        s"""
+          |{
+          |    "accountId": "$accountId",
+          |    "targetDate": "${LocalDate.now.plusYears(1)}",
```

Comment:
To work with an annual sub in all circumstances:
```suggestion
          |    "targetDate": "${LocalDate.now.plusMonths(13)}",
```